NachoSpigot-Server/src/main/java/net/minecraft/server/EntityHuman.java

```diff
@@ -1014,34 +1014,25 @@ public void attack(Entity entity) {
 
                     if (flag2) {
                         if (i > 0) {
-                            KnockbackConfig config = Knockback.get().getConfig();
-                            if(!config.customKnockback) entity.g(-MathHelper.sin(this.yaw * 3.1415927F / 180.0F) * (float) i * 0.5F, 0.1D, (double) (MathHelper.cos(this.yaw * 3.1415927F / 180.0F) * (float) i * 0.5F)); else
+                            KnockbackProfile profile = (entity.getKnockbackProfile() == null) ?
+                                    Nacho.get().getKnockbackConfig().getCurrentKb() : entity.getKnockbackProfile();
                             entity.g(
-                                    (-MathHelper.sin(this.yaw * 3.1415927F / 180.0F) * (float) i * config.knockbackExtraHorizontal),
-                                    config.knockbackExtraVertical,
-                                    (MathHelper.cos(this.yaw * 3.1415927F / 180.0F) * (float) i * config.knockbackExtraHorizontal));
+                                    (-FastMath.sin(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()),
+                                   profile.getExtraVertical(),
+                                    (FastMath.cos(this.yaw * Math.PI / 180.0D) * i * profile.getExtraHorizontal()));
                             this.motX *= 0.6D;
                             this.motZ *= 0.6D;
-                            this.setSprinting(false);
+                            if (profile.isStopSprint()) this.setExtraKnockback(false); //Nacho - Prevent desync player sprinting
                         }
 
                         if (entity instanceof EntityPlayer && entity.velocityChanged) {
-                            // CraftBukkit start - Add Velocity Event
-                            boolean cancelled = false;
                             Player player = (Player) entity.getBukkitEntity();
-                            org.bukkit.util.Vector velocity = new Vector( d0, d1, d2 );
-
-                            PlayerVelocityEvent event = new PlayerVelocityEvent(player, velocity.clone());
+                            PlayerVelocityEvent event = new PlayerVelocityEvent(player, new Vector(entity.motX, entity.motY, entity.motZ));
                             world.getServer().getPluginManager().callEvent(event);
 
                             if (event.isCancelled()) {
-                                cancelled = true;
-                            } else if (!velocity.equals(event.getVelocity())) {
                                 player.setVelocity(event.getVelocity());
-                            }
-
-                            if (!cancelled) {
-                                ( (EntityPlayer) entity ).playerConnection.sendPacket( new PacketPlayOutEntityVelocity( entity ) );
+                                ((EntityPlayer) entity).playerConnection.sendPacket(new PacketPlayOutEntityVelocity(entity));
```

Comment:
Why is this removed?

Reply:
It seems that this skips the cancellation of the player velocity event, not sure why.
@SaithTime Could you explain?

Reply:
t's just a cleanup, the event is not removed

Reply:
> t's just a cleanup, the event is not removed

The default functionality of the event only runs while it is canceled in this code.

Reply:
Just a mistake on my part ^^